DuckDuckGo/Subscription/UserScripts/SubscriptionPagesUseSubscriptionFeature.swift

```diff
@@ -67,10 +67,22 @@ final class SubscriptionPagesUseSubscriptionFeature: Subfeature, ObservableObjec
     struct FeatureSelection: Codable {
         let feature: String
     }
+    
+    enum UseSubscriptionError: Error {
+        case purchaseFailed,
+             missingEntitlements,
+             failedToGetSubscriptionOptions,
+             failedToSetSubscription,
+             failedToRestoreFromEmail,
+             failedToRestoreFromEmailSubscriptionInactive,
+             failedToRestorePastPurchase,
+             subscriptionExpired,
+             hasActiveSubscription,
+             generalError
+    }
         
     @Published var transactionStatus: SubscriptionTransactionStatus = .idle
-    @Published var hasActiveSubscription = false
-    @Published var purchaseError: AppStorePurchaseFlow.Error?
+    @Published var transactionError: UseSubscriptionError?
```

Comment:
(Suggestion, no need to act on it right now)
Storing a var that is accessible from everywhere and everyone is a non-pattern, it's difficult to debug (who set it, who is reading it?) and defeats the overall Swift error handling approach (throw and catch) that allows the error path to be linear.
If an error must be published I would suggest setting the variable only from a single function that is responsible for catching errors from every other function.

Reply:
We need it to be openly readable because of the reason explained above, but I could probably make it private(set), and use a single private function in the class to do that.